ank-core/src/main/kotlin/io/kategory/ank/interpreter.kt

```diff
@@ -142,17 +151,56 @@ fun compileCodeImpl(snippets: Map<File, ListKW<Snippet>>, classpath: ListKW<Stri
                                     it.lang to seManager.getEngineByExtension(extensionMappings.getOrDefault(it.lang, "kts"))
                                 }
                                 .toMap()
+
+                if (codeBlocks.find { it.compile }.nonEmpty()) {
+                    codeBlocks.filter { it.lang != "kotlin" }
+                            .find { it.compile }
+                            ?.let {
+                                throw CompilationException(file, it, IllegalStateException("```${it.lang}:ank:compile is not supported. Use ```${it.lang}:ank or ```${it.lang}:ank:silent"))
```

Comment:
I throw a `CompilationException` here but Gradle seems to get stuck instead of terminating.

Reply:
This is because you are within an async boundary not blocking. If you block the result of `async` it will blow up in the gradle thread.